src/app/clusters/diagnostic-logs-server/diagnostic-logs-server.cpp

```diff
@@ -15,103 +15,251 @@
-    TimeInBufferType timeMs = now.count();
-    chip::ByteSpan payloadTime(reinterpret_cast<uint8_t *>(&timeMs), sizeof(timeMs));
-    return mBuffer.Push(payloadTime, payload);
+    uint16_t ep = emberAfGetClusterServerEndpointIndex(endpoint, Id, EMBER_AF_DIAGNOSTIC_LOGS_CLUSTER_SERVER_ENDPOINT_COUNT);
+    return (ep >= kDiagnosticLogsLogProviderDelegateTableSize ? nullptr : gLogProviderDelegateTable[ep]);
 }
 
-void DiagnosticLogsCommandHandler::InvokeCommand(HandlerContext & handlerContext)
+bool IsLogProviderDelegateNull(LogProviderDelegate * logProviderDelegate, EndpointId endpoint)
 {
-    HandleCommand<chip::app::Clusters::DiagnosticLogs::Commands::RetrieveLogsRequest::DecodableType>(
-        handlerContext, [&](auto & _u, auto & payload) {
-            if (payload.requestedProtocol == chip::app::Clusters::DiagnosticLogs::TransferProtocolEnum::kUnknownEnumValue)
-            {
-                handlerContext.mCommandHandler.AddStatus(handlerContext.mRequestPath,
-                                                         chip::Protocols::InteractionModel::Status::InvalidCommand);
-                return;
-            }
+    if (logProviderDelegate == nullptr)
+    {
+        ChipLogProgress(Zcl, "Diagnosticlogs: no log provider delegate set for endpoint:%u", endpoint);
+        return true;
+    }
+    return false;
+}
+
+} // namespace
+
+DiagnosticLogsServer DiagnosticLogsServer::sInstance;
+
+void DiagnosticLogsServer::SetLogProviderDelegate(EndpointId endpoint, LogProviderDelegate * logProviderDelegate)
+{
+    uint16_t ep =
+        emberAfGetClusterServerEndpointIndex(endpoint, DiagnosticLogs::Id, EMBER_AF_DIAGNOSTIC_LOGS_CLUSTER_SERVER_ENDPOINT_COUNT);
+    if (ep < kDiagnosticLogsLogProviderDelegateTableSize)
+    {
+        gLogProviderDelegateTable[ep] = logProviderDelegate;
+    }
+}
+
+DiagnosticLogsServer & DiagnosticLogsServer::Instance()
+{
+    return sInstance;
+}
+
+#if CHIP_CONFIG_ENABLE_BDX_LOG_TRANSFER
+
+bool DiagnosticLogsServer::IsBDXProtocolRequested(TransferProtocolEnum requestedProtocol)
+{
+    return requestedProtocol == TransferProtocolEnum::kBdx;
+}
+
+bool DiagnosticLogsServer::HasValidFileDesignator(CharSpan transferFileDesignator)
+{
+    return (transferFileDesignator.size() <= kMaxFileDesignatorLen);
+}
+
+CHIP_ERROR DiagnosticLogsServer::HandleLogRequestForBDXProtocol(Messaging::ExchangeContext * exchangeCtx, EndpointId endpointId,
+                                                                IntentEnum intent, CharSpan fileDesignator)
+{
+
+    VerifyOrReturnError(exchangeCtx != nullptr, CHIP_ERROR_INCORRECT_STATE);
+
+    mIntent               = intent;
+    auto scopedPeerNodeId = ScopedNodeId();
+    if (exchangeCtx->HasSessionHandle())
+    {
+        auto sessionHandle = exchangeCtx->GetSessionHandle();
+
+        if (sessionHandle->IsSecureSession())
+        {
+            scopedPeerNodeId = sessionHandle->AsSecureSession()->GetPeer();
+        }
+    }
+
```

Comment:
checking for invalid peer node id and fabric index in InitializeTransfer in DiagnosticLogsBDXTransferHandler as argument check there. it makes more sense since thats the consumer of this data. Please let me know if you disagree

Reply:
I disagree. I think this code should not pass garbage on to other parts of the code @nivi-apple.